src/ansible/roles/keycloak/tasks/main.yml

```diff
@@ -0,0 +1,134 @@
+    url: "{{ keycloak_url }}"
+    dest: /tmp
+    
+- name: Unzip software
+  unarchive:
+    remote_src: yes
+    src: "/tmp/{{ zip_file }}"
+    dest: /tmp
+
+- name: Check if unzipped directory exists
+  stat:
+    path: "{{ unzip_dir }}"
+  register: unzip_dir_chk
+
+- name: debug unzip dir
+  debug:
+    var: unzip_dir_chk.stat
+
+- name: Check if /opt/keycloak dir exists
+  stat:
+    path: /opt/keycloak
+  register: opt_keycloak_chk
+
+- name: debug opt dir
+  debug:
+    var: opt_keycloak_chk.stat
+
+- name: Move software to /opt
+  command: mv {{ unzip_dir }} /opt/keycloak 
+  when: unzip_dir_chk.stat.exists and not opt_keycloak_chk.stat.exists
+
+- name: create Keycloak user/group
+  user:
+    name: keycloak
+    home: /opt/keycloak
+    system: yes
+    create_home: no
+
+- name: change ownership of files in /opt/keycloak
+  file:
+    path: /opt/keycloak
+    owner: keycloak
+    group: keycloak
+    recurse: yes
+
+- name: change ownership of files in /data/certs
+  file:
+    path: /data/certs
+    owner: keycloak
+    group: keycloak
+    recurse: yes
+
+- name: Add CA certificate to keystore
+  shell: |
+    keytool -noprompt -import \
+      -keystore /data/certs/master.keycloak.test.keystore \
+      -file /data/certs/ca.crt \
+      -alias ca.crt \
+      -trustcacerts -storepass Secret123
+  args:
+    creates: /data/certs/master.keycloak.test.keystore
+
+- name: Add Keycloak certificate to keystore
+  shell: |
+    keytool -noprompt -import \
+      -keystore /data/certs/master.keycloak.test.keystore \
+      -file /data/certs/master.keycloak.test.crt \
+      -alias master.keycloak.test.crt \
+      -trustcacerts -storepass Secret123
+
+- name: create keycloak service env file
+  copy:
+    content: |
+      KEYCLOAK_ADMIN=admin
+      KEYCLOAK_ADMIN_PASSWORD=Secret123
+      #KC_LOG_LEVEL=debug
+      KC_HOSTNAME={{ inventory_hostname }}
+      KC_HTTPS_CERTIFICATE_FILE=/data/certs/master.keycloak.test.crt
+      KC_HTTPS_CERTIFICATE_KEY_FILE=/data/certs/master.keycloak.test.key
+      KC_HTTPS_TRUST_STORE_FILE=/data/certs/master.keycloak.test.keystore
+      KC_HTTPS_TRUST_STORE_PASSWORD=Secret123
+      KC_HTTP_RELATIVE_PATH=/auth
+    dest: /etc/sysconfig/keycloak
+
+- name: create systemd keycloak service file
+  copy:
+    content: |
+      [Unit]
+      Description=Keycloak Server
+      After=network.target
+
+      [Service]
+      Type=idle
+      EnvironmentFile=/etc/sysconfig/keycloak
+
+      User=keycloak
+      Group=keycloak
+      ExecStart=/opt/keycloak/bin/kc.sh start
+      TimeoutStartSec=600
+      TimeoutStopSec=600
+
+      [Install]
+      WantedBy=multi-user.target
+    dest: /etc/systemd/system/keycloak.service
+
+- name: Reload systemd to pickup changes added for keycloak
+  systemd:
+    daemon_reload: yes
+
+- name: Run build step for Keycloak and RH-SSO 8+ (skip for RH-SSO 7)
+  shell: |
+    su - keycloak -c '''
+    export KEYCLOAK_ADMIN=admin
+    export KEYCLOAK_ADMIN_PASSWORD=Secret123
```

Comment:
If possible I'd try to avoid defining the secrets and passwords in the tasks and do that in the [proper file](https://github.com/SSSD/sssd-ci-containers/blob/master/src/ansible/group_vars/all#L7)